Chime/Parser.m

```diff
@@ -193,9 +238,51 @@ - (void)handleInstructions:(BOOL)isLabel
         exceptionWithName:@"Invalid line encountered"
                    reason:[NSString stringWithFormat:
                                         @"\nLine %@:\n \"%@\" is invalid",
-                                        @(_counter), string]
+                                        @(_physicalLinesCounter), string]
+                 userInfo:nil];
+  }
+}
+
+NSMutableArray *packString(NSString *string) {
+  NSMutableArray *pack = [[NSMutableArray alloc] init];
+  NSUInteger counter = 0;
+  uint32_t buffer = 0;
+  NSUInteger wordSize = 4;
+  NSData *stringData = [string dataUsingEncoding:NSUTF8StringEncoding];
+  const char *bytes = [stringData bytes];
+  for (int i = 0; i < [stringData length]; i++) {
+    buffer = buffer << 8;
+    buffer |= (char)bytes[i];
+    counter++;
+    if (counter == wordSize) {
+      [pack addObject:@(buffer)];
+      buffer = 0;
+      counter = 0;
+    }
+  }
+  if (counter != 0) {
+    [pack addObject:@(buffer << (8 * (wordSize - counter)))];
+  }
+
+  return pack;
+}
+
+- (void)handleStringLine:(NSString *)chimeString
+                    fill:(NSMutableArray *)bytecodes {
+  NSString *content = popBothEnds(chimeString);
+  NSUInteger length = content.length; // TODO: Escape codes
+  NSMutableArray *stringPayload = packString(content);
+  if (length > MAX_STRING_SIZE) {
+    @throw [NSException
+        exceptionWithName:@"String limit exceeded"
+                   reason:[NSString
+                              stringWithFormat:
+                                  @"String \"%@\" exceeded max size of %d",
+                                  content, MAX_STRING_SIZE]
                  userInfo:nil];
   }
+  [bytecodes addObject:@((uint32_t)length)];
+  [bytecodes addObjectsFromArray:[stringPayload copy]];
```

Comment:
Any particular reason to copy the payload here?

Reply:
I believe this has something to do with converting `NSMutableArray` to `NSArray`.